src/annotator/integrations/test/html-test.js

```diff
@@ -354,4 +366,66 @@ describe('HTMLIntegration', () => {
       assert.deepEqual(await integration.uri(), 'https://example.com/');
     });
   });
+
+  it('emits "uriChanged" event when URL changes after a navigation', () => {
+    const integration = createIntegration();
+    const onURIChanged = sinon.stub();
+    integration.on('uriChanged', onURIChanged);
+
+    fakeHTMLMetadata.uri.returns('https://example.com/new-url');
+    notifyNavigation();
+
+    assert.calledWith(onURIChanged, 'https://example.com/new-url');
+  });
+
+  it('emits "uriChanged" event when URL changes after a <head> change', async () => {
+    const linkEl = document.createElement('link');
+    linkEl.rel = 'dummy';
+    linkEl.href = 'https://example.com/dummy';
+
+    const integration = createIntegration();
+    const onURIChanged = sinon.stub();
+    integration.on('uriChanged', onURIChanged);
+
+    try {
+      document.head.append(linkEl);
+      fakeHTMLMetadata.uri.returns('https://example.com/new-url');
+
+      await delay(0); // Wait for MutationObserver
+
+      assert.calledWith(onURIChanged, 'https://example.com/new-url');
+    } finally {
+      linkEl.remove();
+    }
+  });
+
+  it('does not emit "uriChanged" if URL has not changed after a navigation', () => {
+    const integration = createIntegration();
+    const onURIChanged = sinon.stub();
+    integration.on('uriChanged', onURIChanged);
+
+    notifyNavigation();
+
+    assert.notCalled(onURIChanged);
+  });
```

Comment:
Is this because `fakeHTMLMetadata.uri()` returns the same thing? It's a little hard to see what makes this test "work", that is.

Reply:
I added a comment here to explain what is going on.